### DISCUSSION TITLE
[Help] Share your TTS models

### DISCUSSION THREAD

**Discussion by erogol** [coqui-ai/TTS#930] [General]
Please consider sharing your pre-trained models in any language (If the licences allow that). 

We can include them in our model catalogue for public use by attributing your name (website, company etc.).

That would enable more people to experiment together and coordinate, instead of individual efforts to achieve similar goals.

That is also a chance to make your work more visible. 

You can share in  two ways; 

1. Share the model files with us and we serve them with the next :frog: TTS release.
2. Upload your models on GDrive and share the link. 

Models are served under ```.models.json``` file and any model is available under ```tts``` CLI or Server end points. [More details...](**url**)

(previously  mozilla/TTS#395)

**Comment by enjikaka** [comment]
Any ELI5 tutorial/doc for creating a dataset for your own language/dialect?

**Comment by erogol** [comment]
Not sure if it is ELI5, but there is this link https://github.com/coqui-ai/TTS/wiki/What-makes-a-good-TTS-dataset

Also, @thorstenMueller has created a TTS dataset from the gecko so he might have valuable comments if you have specific questions.

**Comment by thorstenMueller** [comment]
Feel free to ask specific question. I'd happy to share my experiences on recording a new dataset here.

* Find/Create a text corpus to record (one sentence = 1 recording)
* Replace numbers to text
* Create csv file from corpus
* Check Mimic-Recording-Studio from Mycroft as recording environment (https://github.com/MycroftAI/mimic-recording-studio)
* Start recording
  * Constant speed while recordings
  * Speak all chars clearly
  * Speak in neutral voice
  * Use good microphone equipment
  * Find a recording place without random noise

**Reply by hjschiller** [reply]
> Feel free to ask specific question. I'd happy to share my experiences on recording a new dataset here.
> 
> * Find/Create a text corpus to record (one sentence = 1 recording)
> * Replace numbers to text
> * Create csv file from corpus
> * Check Mimic-Recording-Studio from Mycroft as recording environment (https://github.com/MycroftAI/mimic-recording-studio)
> * Start recording
>   
>   * Constant speed while recordings
>   * Speak all chars clearly
>   * Speak in neutral voice
>   * Use good microphone equipment
>   * Find a recording place without random noise

I am a native English (American) speaker, male, but high pitched (so often confused for a woman in the phone). I am just starting out in making a model and am happy to share my model if that is helpful. So far I have only attempted a single recording session, about 700 sentences, in not the quietest location. 

My worst recordings have a SNR of 20 and average around 34 (but I just realized none of the noise reduction I applied actually worked so I plan to work on that while recovering from COVID). 

What is the most effective least time involved way to test various dataset creation methods? For example would reading 3 sentences with two different mics be sufficient for me (or someone) to judge which provides “better” results? What would the metric be? Signal to noise ratio?

For the dataset, what exactly do you mean? I assume this is .wav files and metadata.csv, but do you want the raw recording, the recording with extra dead air removed, band pass filtered to take out pops, noise reduced, compressed, amplified to a maximum of -10db?  Do you have a decent write up (in English) of a good workflow for making a good data set? (A YouTube video would be ok too)

Also I’m not exactly sure what neutral voice is, I have been aiming for the voice I use in conversation, or reading a book out-loud (without doing different voices for when the characters speak). It is not necessarily flat, but it is without emotion such as excitement or anger.

**Comment by Sadam1195** [comment]
Hi @erogol , thank you for the amazing work, from Mozilla TTS to coqui-ai. Although Mozilla seemed perfect to me as it had wider community reach, just hope this grows even wider and faster than Mozilla. I am planning to share my models for Spanish and Italian using (**Taco2 600k steps + WaveRNN**). Audio quality seems to be good but I need to train it a bit more and also ask dataset providers if that would be okay if I make the models public. 
Fingers crossed. 


Let me know if I can contribute in any way I have Google Colab Pro resources laying around free.


+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.67       Driver Version: 460.32.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:00:04.0 Off |                    0 |
| N/A   35C    P0    24W / 300W |      0MiB / 16160MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

**Comment by erogol** [comment]
@Sadam1195 thx for the amazing work :rocket::rocket:.  

I really hope we can include your models, of course with the right attribution going to you. 

Just waiting for your signal.

For general contribution, this is a nice place to start https://github.com/coqui-ai/TTS/blob/main/CONTRIBUTING.md

If you just like to train models, let me know we can also find new datasets to attack.

**Comment by Sadam1195** [comment]
> I really hope we can include your models, of course with the right attribution going to you.
> **I hope they allow me, otherwise I would see it as wasting my time and effort.**
> Just waiting for your signal.
> **I will let you know when I get the confirmation**. 
> If you just like to train models, let me know we can also find new datasets to attack.
**Training models on colab can be a bit annoying as sessions often get disconnected even with all the tricks in the book.**

Nonetheless, I would love to train model on new datasets (if you have any) specially in the languages in which TTS models haven't been made public yet.

**Comment by kaiidams** [comment]
Hello,

I've just started to train a public domain Japanese dataset https://github.com/kaiidams/Kokoro-Speech-Dataset with Tacotron 2 of the latest master of https://github.com/mozilla/TTS on Google Colab Free. After 19K steps, I can hear what he says, although it is metallic.

To proceed, I'd like to know which branch and repo do you recommend for me to use? https://github.com/erogol/TTS_recipes seems a bit old.

**Comment by Sadam1195** [comment]
> To proceed, I'd like to know which branch and repo do you recommend for me to use? https://github.com/erogol/TTS_recipes seems a bit old.

Please use this https://github.com/coqui-ai/TTS instead of  https://github.com/mozilla/TTS and use the latest main branch. @kaiidams

**Comment by kaiidams** [comment]
@Sadam1195 @erogol 

I trained Tacotron 2 for 130K steps with this code https://github.com/kaiidams/TTS/tree/kaiidams/kokoro which was forked from the latest main.
https://drive.google.com/drive/folders/1-1_HB-ogmvD-qYaHm8D5Xp1pWq9HKhB_?usp=sharing 
The included sample.wav was generated with vocoder_models/universal/libri-tts/wavegrad.

The input of the model is Romanized Japanese text. It requires some dependencies like MeCab to convert texts from ordinary ones.
The dataset is the public domain and the reader knows about the dataset. I think I can provide Python code for text conversion.

**Comment by erogol** [comment]
@kaiidams if you can send a PR for text conversion something similar to the Chinese API we have, with the model, would be a great contribution.

**Reply by Grant-Tao** [reply]
Sorry to bother. Is it possible to add a chinese dataset when training yourtts vits model, so that yourtts can speak chinese? if yes, how?

**Comment by zubairahmed-ai** [comment]
> Feel free to ask specific question. I'd happy to share my experiences on recording a new dataset here.
> 
> * Find/Create a text corpus to record (one sentence = 1 recording)
> * Replace numbers to text
> * Create csv file from corpus
> * Check Mimic-Recording-Studio from Mycroft as recording environment (https://github.com/MycroftAI/mimic-recording-studio)
> * Start recording
>   
>   * Constant speed while recordings
>   * Speak all chars clearly
>   * Speak in neutral voice
>   * Use good microphone equipment
>   * Find a recording place without random noise

Any reason why this and [this](https://github.com/coqui-ai/TTS/issues/380#issuecomment-820468466) isn't in the readme? 
I had to look up training to reach here

**Comment by thorstenMueller** [comment]
Hi @zubairahmed-ai.
Here's a talk a made on how to record a voice dataset if that's helpful for you.

https://youtu.be/m-Uwb-Bg144

**Comment by zubairahmed-ai** [comment]
@thorstenMueller Perfect timing, thank you

**Comment by zubairahmed-ai** [comment]
Oh just realized this talk happened during recent Google I/O and I somehow didn't catch it while watching other videos :)

**Comment by zubairahmed-ai** [comment]
@thorstenMueller Thanks so much for the great video explaining your process in details with some tips. I'll make sure I follow that, do you plan to give a try to other models besides Tacotron-2? like Align-TTS?

**Comment by thorstenMueller** [comment]
You're welcome @zubairahmed-ai :-).
I'm currently finishing some recording stuff for my emotional dataset and train a Fullband-MelGAN vocoder. So i've no time left to look at other models like Align-TTS. But feel free to train a "Thorsten" model with Align-TTS ;-).

**Comment by stale** [comment]
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions. You might also look our discussion channels.

**Comment by ravi-maithrey** [comment]
Asking people to share their models can also be added to the CONTRIBUTING.md, since it is asking for contributions. I'd be up to doing that, if no one has taken it up yet?

**Comment by erogol** [comment]
yeah good point. Feel free to take on it.

**Comment by ghost** [comment]
I would like to contribute my own model.. but I stuck in middle.. I have created dataset(LJSpeech) of my own voice . For training my model I need config.json file , so can anyone provide me the template of config.json file for LJSpeech dataset format required to train my model.

Thanks in Advance

**Comment by erogol** [comment]
@ManoBharathi93  you can start from the LJSpeech recipes in the `recipes` folder and change the config fields for your dataset specs. You can find more info here https://tts.readthedocs.io/en/latest/

**Comment by ghost** [comment]
@erogol  thanks a lot sir

**Comment by ghost** [comment]
Hello folks, How can I add drop-down Menu to list available models(downloaded models) in WEB-UI and when I change the server.py file the web interface is not changing ? please mention which file name want to make changes impact in WEB-UI..

**Comment by godspirit00** [comment]
I'd like to share a Tacotron2-DCA model and a Univnet model I trained on the Nancy corpus.

Here is a sample:

https://user-images.githubusercontent.com/16176151/136530668-4b037ea5-8b26-4543-8e89-072f04f9c658.mp4

The link to the models:
https://drive.google.com/drive/folders/1bMNOjjYxcCkgwkcYAlsPR3qM4hZQzAOR?usp=sharing

Thanks again for the great work!

**Reply by ficoquery** [reply]
What was your recording hardware, voice editing software configuration for this?

**Reply by godspirit00** [reply]
This is a dataset you can simply download for free from [here](https://www.cstr.ed.ac.uk/projects/blizzard/2011/lessac_blizzard2011/) .
I didn't record it myself, but I did do the splitting of the 44k wave files and downsample them to 22k.

**Reply by wagesj45** [reply]
I downloaded the model and a few places had that robotic/metalic twinge, whereas your sample was completely natural sounding. Was there something specific you did that gave that perfect result? Changed settings when running the model, or post processing maybe?

**Reply by tobiccino** [reply]
Hi @godspirit00 ,  im training univnet vocoder from scratch(Vietnamese language) on a dataset have 24 hours of voice and facing this problem. At first evaluation my avg_G_adv_loss is very high -> avg_G_adv_loss : 15928888607637847932928.00000. Did you facing that problem at first step, thanks sm.

**Comment by erogol** [comment]
@godspirit00 the quality is awesome.

**Comment by gullabi** [comment]
I am about to train a TTS model in Catalan and I have a question concerning the g2p processes. Since gruut does not support Catalan, I checked the similar cases of Ukrainian and Japanese. As far as I understand I would need to either use graphemes for training or write a phonemizer for Catalan. In principle, espeak has Catalan, but I don't know whether it is best to include also espeak within the repo. Is there a strategy/policy concerning the languages which are in this awkward in between situation, i.e. languages that have latin script but not supported by gruut.

**Reply by erogol** [reply]
Actually, I am changing the text processing with a complete rewrite. It'll have espeak enabled as an option so you can probably use Catalan phonemizer. But I need a bit of time, probably until the end of the week. Once done, you can try out the PR branch.

**Reply by gullabi** [reply]
Sounds great! I will keep an eye on the commits.

**Comment by gullabi** [comment]
Hi all! [Here](https://drive.google.com/drive/folders/1zhcAWd-YR0S9-kQN2TJAlee7hFOKiFgH) are the Catalan text-spec models that I trained. The models come with some caveats:

* Unfortunately, I don't have the vocoder models yet, I ran out of computational resources during the training and couldn't converge to a good quality. 
* For some reason, the models work with the dev branch only. In the main branch, I am getting `None` when I call `setup_model`.
* Most importantly, there also are some normalization and pronunciation issues, which will be solved with the introduction of the #937 PR
* Finally in my own fork, I wrote functions for `catalan_cleaners`, but I didn't do a PR yet, since I am waiting for the #937 PR to be merged first. For portability, I put `basic_cleaners` in the configs, and they should work however with some corner-case pronunciation issues. 

Here are the results: Ona synthesized with GL

https://user-images.githubusercontent.com/40303490/145807369-8758fb5a-208c-412e-b56f-ae160e9fc686.mp4

Pau with GL

https://user-images.githubusercontent.com/40303490/145807697-3cb7be21-09d6-4fee-ac4c-9eaa12d46914.mp4

Pau with [descriptinc/melgan-neurips](https://github.com/descriptinc/melgan-neurips)

https://user-images.githubusercontent.com/40303490/145807722-be6ee4ee-0664-4bb5-a8a3-d8002767af1a.mp4

Sorry for the caveats, I just wanted to get the models out for anyone to start testing them. Also I will update the [catotron](https://github.com/CollectivaT-dev/catotron-cpu) repo for easy synthesis with the vocoder.

**Reply by erogol** [reply]
The prosody is really good for the first audio. I think with a vocoder it'd sound great. 

If there are changes for the #937 feel free to send a ✨PR✨ to the branch. It will take some time until we merge these API changes.

**Reply by gullabi** [reply]
Thanks Eren! Now I have to "ship" the new Catotron, since we are working on a calendar. So no worries about the merge, and I will try to take a look when we wrap up the project.

**Comment by ZohaibSajjad** [comment]
Hi all, I want to build voice conversion using the cross-language technique. For this purpose, I have used voice conversion challenge 2020 architecture. After 6 weeks of working on it, there are no acceptable results found. Now I decide to use YourTTS voice conversion architecture. I want to train it on the English-Urdu dataset. But I don't know where to start, can anyone guide me in this regard, Any help is appreciated.

**Comment by xettrisomeman** [comment]
Hey , I  would like to share my Nepali Model trained on [Openslr dataset](https://www.openslr.org/43/)

demo : [vitsexamplex2_1015k.webm](https://user-images.githubusercontent.com/44607279/188865427-279c5154-0d26-4388-b454-4be8e6b6801f.webm)

Here is the drive link: https://drive.google.com/drive/folders/1Jwr7ITDA4hFKLMSVXUj3A8nQlpdxT6ql?usp=sharing

Thanks :)

**Reply by erogol** [reply]
Thanks for sharing. Does it work with the latest 🐸TTS?

**Comment by Tarek-Hasan** [comment]
Checkout Voice models for Mimic 3 from Mycroft AI
https://github.com/MycroftAI/mimic3
these are the most natural sounded opensource AI voice I ever heard.

**Comment by erogol** [comment]
@Tarek-Hasan looking at the released models they pretty much used our code but I don't see the released model binaries.

**Comment by Tarek-Hasan** [comment]
Sorry, I somehow included wrong link. Here's the voice models link
https://github.com/MycroftAI/mimic3-voices

**Reply by erogol** [reply]
It looks like they only released ONNX files that are not unfortunately compatible with TTS. Some manual work is needed.
Thanks for sharing the link though 👍

**Comment by gullabi** [comment]
I would like to present a new VITS multispeaker model trained by @GerrySant for Catalan within the framework of @projecte-aina. It is trained from zero with 101460 utterances consisting of 257 speakers, approx 138 hours of speech. We used three datasets; Festcat and Google Catalan TTS (both TTS datasets) and also a part of Common Voice 8. It is trained with TTS v0.8.0.

Here are two examples of a male and a female voice.

https://user-images.githubusercontent.com/40303490/208879552-ea0c0706-7c25-4560-a00e-6246544b0109.mp4

https://user-images.githubusercontent.com/40303490/208879565-6b60e744-fd6f-4df5-9441-d4497887a61e.mp4

The model is uploaded in [Huggingface](https://huggingface.co/projecte-aina/tts-ca-coqui-vits-multispeaker) with its own [space](https://huggingface.co/spaces/projecte-aina/tts-ca-coqui-vits-multispeaker) to generate voices. We also would like the models to be accessible as a part of Coqui models.

**Reply by erogol** [reply]
@gullabi added the model to the latest release https://github.com/coqui-ai/TTS/releases/tag/v0.10.1_models

Just one note. You need to rename `speakers.pth` as `sepeaker_ids.pth` to be compatible with the ```tts ``` command. 

`speakers.pth` is for models trained with d-vectors. 

Thanks again for sharing the model.

**Comment by erogol** [comment]
@gullabi thanks for sharing. I'll add the model asap.

**Comment by karim23657** [comment]
I trained a glow_tts model for Persian language.
https://huggingface.co/Kamtera/persian-tts-female-glow_tts
Also I have created a dataset:
https://www.kaggle.com/datasets/magnoliasis/persian-tts-dataset-famale

**Reply by erogol** [reply]
Thanks for sharing. A new language 👍

**Reply by erogol** [reply]
@karim23657 model audio quality is not very good since there is no vocoder. To fix that here are my suggestions

- Retrain/finetune the model with the audio parameters that are compatible with one of the released universal vocoders
- Train a separate vocoder model compatible with the model
- Train VITS instead of GlowTTS, which has a vocoder inside. 

Let me know if you are interested in following one of the options. It'd be nice to bump up the model quality to make it useful. 

PS. I've added the model to the latest releases regardless.

https://github.com/coqui-ai/TTS/releases/tag/v0.10.1_models

**Reply by karim23657** [reply]
I trained these VITS :
https://huggingface.co/Kamtera/persian-tts-male-vits
https://huggingface.co/Kamtera/persian-tts-female-vits

Demo:
https://huggingface.co/spaces/Kamtera/Persian-tts-CoquiTTS

Models trained on these datasets : 
https://www.kaggle.com/datasets/magnoliasis/persian-tts-dataset 
https://www.kaggle.com/datasets/magnoliasis/persian-tts-dataset-famale

Other models(not very good): https://huggingface.co/Kamtera

**Comment by mobassir94** [comment]
@erogol 
i would like to share two bangla tts models trained using fantastic 🐸 TTS.
i have trained vits and glowtts,, though glowtts didn't perform well but vits performed really well. i have used iitm IndicSpeech dataset and converted them in ljspeech format for training the models using 🐸 TTS api.
please have a look at this repo : https://github.com/mobassir94/comprehensive-bangla-tts 
everything is public,including the trained weight files and logs. please check this colab inference demo code : https://github.com/mobassir94/comprehensive-bangla-tts/blob/main/Comprehensive_Bangla_Text_to_Speech_(TTS).ipynb

To the best of our knowledge (from our extensive google search and research and extensive human validation) we’ve discovered that the Bangla Vits TTS (text to speech) system that we trained and used for reading various bangla tafsir / hadith is the highest performing State of the Art (SOTA) Bangla neural voice cloning system till this date (Thursday, December 29, 2022) that’s ever released publicly for Bangla language for free and it beats past TTS systems like gtts,silero-tts,indic-tts by large margin in terms of quality.
i would like to thank 🐸 TTS team. your api helped me to train and deploy best bangla tts models with ease.

**Reply by erogol** [reply]
It's been a while but I did not forget about it. Hopefully this week I'll give your model a try and hopefully add it to the model zoo.

**Reply by mobassir94** [reply]
@erogol bangla is a very complex language and there are many complex issues with this languages,,if you check this notebook -> https://github.com/mobassir94/comprehensive-bangla-tts/blob/main/Comprehensive_Bangla_Text_to_Speech_(TTS).ipynb you can see that i used import bangla,from bnnumerizer import numerize
 and from bnunicodenormalizer import Normalizer 
 
 these packages needs to be used to process the input text before fetching it to tts model for inference.these pre processings are very necessary

**Reply by erogol** [reply]
it was actually quite easy. I am adding the female and male models to the model zoo. I'll ping you to try it out.

**Reply by erogol** [reply]
models are gone through the release 🚀

**Comment by neural-loop** [comment]
Started a project for creating collections of voices: https://huggingface.co/voices/

The first addition is https://huggingface.co/voices/VCTK_European_English_Females (trained at 85000 steps, might go a bit further later) A nice feature is the ability to quickly preview each of the voices based on the wav samples / README markdown.

I plan to go through and include all of the VCTK voices. It would be kind of cool to do one big VCTK dataset with the 110 english voices, but it would be so many dataset files that it'd feel like a bit of a slog to get through, and also I notice that the voices kind of pull each other different directions (in the VCTK European English Females, for example, the original YourTTS male-en-2 has taken on the properties of the female voices) so I considered that segmenting them by similarity might produce better outputs. 

A couple other community members in the discord have been helping, one is setting up a space where people can test the voices out and another tested the model and helped me iron out some issues in the config.json (which I think may still need a little work)

These can be added to the model zoo or list if wanted. The VCTK set will be 100% from VCTK & YourTTS training data so it should all be 'for sure' CC-by-4.0

I may add some custom curated voices that sit more in the gray area to the voices space on huggingface but it will be labeled clearly which ones were trained off of cc-by-4.0 or if they have manually sourced training.

**Comment by kungfooman** [comment]
Anyone happens to have a Norwegian voice model?

**Comment by vislupus** [comment]
I prepared a small [dataset](https://github.com/vislupus/Bulgarian-TTS-dataset) in a format similar to LJSpeech for Bulgarian and English. I can also add more audio files for an additional speaker if that will be helpful.

**Reply by dieselcenter** [reply]
Can you help for bg model training ?

**Comment by alex73** [comment]
Hi. I would like to share model for Belarusian language trained on the Mozilla Common Voice dataset. License: CC-BY-SA 4.0
Please share it from https://coqui.gateway.scarf.sh/
Files for GlowTTS and HiFiGAN : https://drive.google.com/drive/folders/19OmP81aeOd2Xp7aP8aVSxhuXnR4s43QO?usp=sharing

**Reply by erogol** [reply]
Thanks for sharing the model. I'll add it to the model zoo ASAP

**Reply by erogol** [reply]
now i am on it... sorry for the delay.

**Reply by erogol** [reply]
@alex73 I need this file   
 
"stats_path": "/storage/TTS/scale_stats.npy"

**Reply by alex73** [reply]
I placed it to https://drive.google.com/drive/folders/19OmP81aeOd2Xp7aP8aVSxhuXnR4s43QO. IMHO, synthesis doesn't need it - just for load configs, probably.

**Reply by erogol** [reply]
Can you share a sample output from the model? I couldn't get something reasonable. Maybe I am doing something wrong. 

Also, I tried using Belarusian translation of a text from Google Translate but most of the characters are not in the vocabulary. I am not sure what input I should use with this model. Do you input transliterated text?

**Reply by alex73** [reply]
See the output: https://drive.google.com/file/d/1OQDVM3FE8QmiiOTkC8bGb1llQWG1nCb5/view?usp=drive_link
It requires phonemes as IPA string for input("ˈvasʲ vʲiˈtajɛ sʲintɛˈzatar bʲɛɫaˈruskaɣa mau̯ˈlʲɛnʲːa!" in our case). 
You can convert text to phonemes on the https://bnkorpus.info/fanetyka.html
BTW, next question is about phonemizer. Coqui-TTS uses gruut, espeak, and some custom phonemizers.
We have java phonemizer and there is Py4J (and other frameworks to integrate python and java).
So, what will be best way ? To integrate our phonemizer directly info Coqui-TTS using Py4J, or integrate it into some other framework (like gruut or espeak), then use via such frameworks ?

**Reply by erogol** [reply]
You should add it as a separate backend like espeak and worse case call it from the terminal again like espeak. You can use the espeak implementation as an example under `TTS/TTS/tts/utils/text`

**Reply by erogol** [reply]
Model sounds really good btw :)

**Reply by alex73** [reply]
@erogol Hi Eren. Did you include Belarusian model into some official list ? I don't see it in the "--list_models".

**Reply by erogol** [reply]
not yet I needed the phonemizer. It is just merged. I'll add the model asap. Sorry, I am slow working on open-source among other things in the list.

**Reply by erogol** [reply]
https://github.com/coqui-ai/TTS/pull/2922

**Comment by karim23657** [comment]
Bellow repo contains my Models and demoes and training codes for Persian tts 

https://github.com/karim23657/Persian-tts-coqui

**Reply by erogol** [reply]
Thanks for sharing. I'll try to add them to the model zoo asap.

**Reply by erogol** [reply]
@karim23657 among all the models do you have a favorite model? It'd be better to add the best model only so people do not need to jump between models.

**Reply by karim23657** [reply]
@erogol  these are  best : [vits female (best)](https://huggingface.co/Kamtera/persian-tts-female-vits) and [vits male1 (best)](https://huggingface.co/Kamtera/persian-tts-male1-vits)

**Reply by erogol** [reply]
Thanks I'll give them a try asap

**Reply by UncleBob2** [reply]
https://huggingface.co/spaces/ntt123/Vietnam-male-voice-TTS

Could you please add this Vietnamese model on Hugging Face?

**Comment by omega3** [comment]
Can you share some good British English model voice? US English voices like  17: tts_models/en/ljspeech/vits--neon is great but British voices are not as good. As a point of reference I propose voices from a [dictionary](https://www.ldoceonline.com/dictionary/advantage).

**Comment by Poupou2002** [comment]
Hi! I’m exploring various TTS models and found that using AudioModify helps improve the audio quality by reducing robotic artifacts after synthesis. If anyone’s interested, I can share tips on combining it with different models. Thanks!
